Creando Arrays

In [1]:
import numpy as np

In [2]:
edad_miembros=np.array([22,19,24,28,26,30,23])

In [3]:
print('Edad de los miembros del equipo de futbol:', edad_miembros)

Edad de los miembros del equipo de futbol: [22 19 24 28 26 30 23]


In [4]:
dias_abiertos=np.arange(1,31) #Asuminedo que un mes tenga 30 dias
print('Dias en que el club está abierto:',dias_abiertos)

Dias en que el club está abierto: [ 1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24
 25 26 27 28 29 30]


In [5]:
puntos_tiempo=np.linspace(0,60,10) #De 0 a 60 minutos, 10 intervalos
print('Puntos de tiempo durante una hora de entrenamiento:', puntos_tiempo)

Puntos de tiempo durante una hora de entrenamiento: [ 0.          6.66666667 13.33333333 20.         26.66666667 33.33333333
 40.         46.66666667 53.33333333 60.        ]


In [6]:
asistencia_natacion=np.random.rand(7) > 0.5 # Suponiendo una probabilidad del 50%
print('Asistencia a la clase de natación durante una semana:', asistencia_natacion)

Asistencia a la clase de natación durante una semana: [False False False  True False  True False]


In [7]:
mejora_rendimiento = np.random.randn(5)  # 5 nadadores

print("Mejora en el rendimiento de los nadadores (tiempos de vuelta):", mejora_rendimiento)

Mejora en el rendimiento de los nadadores (tiempos de vuelta): [-1.90503069  0.18676912 -0.1750589   0.07379586 -2.84516032]


Uso de shape

In [8]:
# Registro de asistencia: filas representan deportes [Natación, Tenis],

# columnas representan días de la semana [Lunes a Viernes]

asistencia = np.array([

    [1, 0, 1, 1, 0],  # Natación

    [0, 1, 1, 0, 1]   # Tenis

])


print("Registro de asistencia:\n", asistencia)

Registro de asistencia:
 [[1 0 1 1 0]
 [0 1 1 0 1]]


In [9]:
print("Shape del array de asistencia:", asistencia.shape)

Shape del array de asistencia: (2, 5)


Manipulacion de Arrays
Uso de reshape

array.reshape(newshape*): cambia la forma de un array sin cambiar sus datos. 

In [10]:
# Registro de asistencia para 2 semanas de clases de natación (10 días)
asistencia_natacion = np.array([1, 0, 1, 1, 0, 1, 1, 0, 1, 1])

print("Asistencia original:", asistencia_natacion)

Asistencia original: [1 0 1 1 0 1 1 0 1 1]


In [11]:
# Cambiando la forma del array para que tenga 2 filas (semanas) y 5 columnas (días)
asistencia_reshaped = asistencia_natacion.reshape(2, 5)

print("Asistencia reorganizada (2 semanas, 5 días):\n", asistencia_reshaped)

Asistencia reorganizada (2 semanas, 5 días):
 [[1 0 1 1 0]
 [1 1 0 1 1]]


Creando una copia

In [12]:
# Creando una copia del array de asistencia reorganizado
asistencia_copiada = asistencia_reshaped.copy() # el nombre original del archivo es asistencia_reshaped

# Simulamos una corrección en los datos: cambiamos la asistencia del primer día de la segunda semana
asistencia_copiada[1, 0] = 0

print("Asistencia copiada y modificada:\n", asistencia_copiada) #La copia deberia llamarse asistencia_reshaped_copiada para mayor claridad
print("Asistencia original sin cambios:\n", asistencia_reshaped)

Asistencia copiada y modificada:
 [[1 0 1 1 0]
 [0 1 0 1 1]]
Asistencia original sin cambios:
 [[1 0 1 1 0]
 [1 1 0 1 1]]


Operaciones con Arrays

Suma (+)

In [13]:
horas_natacion = np.array([1, 2, 1, 1.5, 2])  # Horas por día de lunes a viernes
horas_tenis = np.array([1.5, 2, 2, 1, 1.5])
total_horas = horas_natacion + horas_tenis

print("Total de horas gastadas en deportes:", total_horas)

Total de horas gastadas en deportes: [2.5 4.  3.  2.5 3.5]


Resta (-)

In [14]:
diferencia_horas = horas_natacion - horas_tenis
print("Diferencia de horas entre natación y tenis:", diferencia_horas)

Diferencia de horas entre natación y tenis: [-0.5  0.  -1.   0.5  0.5]


Multiplicación (*)

In [15]:
ajuste_natacion = horas_natacion * 1.1
ajuste_tenis = horas_tenis * 1.2

print("Horas ajustadas por intensidad (Natación):", ajuste_natacion)
print("Horas ajustadas por intensidad (Tenis):", ajuste_tenis)

Horas ajustadas por intensidad (Natación): [1.1  2.2  1.1  1.65 2.2 ]
Horas ajustadas por intensidad (Tenis): [1.8 2.4 2.4 1.2 1.8]


In [17]:
#Probando epor cuenta propia multiplicacion entre arrays
multiplicacion_horas = horas_natacion * horas_tenis
print(multiplicacion_horas)

[1.5 4.  2.  1.5 3. ]


División (/)

In [18]:
promedio_natacion = total_horas / 5  # Suponiendo 5 días a la semana
print("Promedio de horas por deporte por día:", promedio_natacion)

Promedio de horas por deporte por día: [0.5 0.8 0.6 0.5 0.7]


In [19]:
#Probando entre arrays
division_horas= horas_natacion / horas_tenis #No tiene sentido, solo como prueba
print(division_horas)

[0.66666667 1.         0.5        1.5        1.33333333]


Multiplicaion de matrices
np.matmul

In [20]:
# Habilidades del equipo por deporte [natación, tenis]
habilidades = np.array([[0.8, 0.6],  # Eficiencia
                        [0.9, 0.95]])  # Técnica

# Efectividad de entrenamiento [Eficiencia, Técnica]
efectividad_entrenamiento = np.array([[1.2, 0.8],  

                                      [1.1, 1.0]])

# Cálculo del rendimiento total
rendimiento_total = np.matmul(habilidades, efectividad_entrenamiento)
print("Rendimiento total por deporte:", rendimiento_total)

Rendimiento total por deporte: [[1.62  1.24 ]
 [2.125 1.67 ]]


In [23]:
print(habilidades)
print(efectividad_entrenamiento)
probando_multiplicacion_tradicional=habilidades*rendimiento_total
print(probando_multiplicacion_tradicional) #Es diferente no funciona como multiplicacion de matrices

[[0.8  0.6 ]
 [0.9  0.95]]
[[1.2 0.8]
 [1.1 1. ]]
[[1.296  0.744 ]
 [1.9125 1.5865]]


# Habilidades del equipo por deporte [natación, tenis]
habilidades = np.array([[0.8, 0.6],  # Eficiencia
                        [0.9, 0.95]])  # Técnica

# Efectividad de entrenamiento [Eficiencia, Técnica]
efectividad_entrenamiento = np.array([[1.2, 0.8],  

                                      [1.1, 1.0]])

# Cálculo del rendimiento total
rendimiento_total = np.matmul(habilidades, efectividad_entrenamiento)
print("Rendimiento total por deporte:", rendimiento_total)

In [29]:
a = 0.8 * 1.2
b = 0.6 * 1.1
print(a, b, a + b)


0.96 0.66 1.62


Aquí se muestra un cálculo individual:

𝑎=0.8×1.2=0.96
a=0.8×1.2=0.96
𝑏=0.6×1.1=0.66
b=0.6×1.1=0.66
𝑎+𝑏=0.96+0.66=1.62
a+b=0.96+0.66=1.62
Esto corresponde a una de las entradas en el resultado de la matriz.

Indexación, slicing y reshaping
array[idx] → Indexación

In [24]:
edades_natacion = np.array([20, 22, 19, 21, 23, 24])
# Accediendo al tercer miembro del equipo
edad_tercer_miembro = edades_natacion[2]  # Recuerda que la indexación comienza en 0
print("Edad del tercer miembro:", edad_tercer_miembro)

Edad del tercer miembro: 19


array[start:stop:step] → Slicing 

In [25]:
# Seleccionando las edades de los primeros tres miembros
primeras_tres_edades = edades_natacion[0:3]  # O edades_natacion[:3]
print("Edades de los primeros tres miembros:", primeras_tres_edades)

Edades de los primeros tres miembros: [20 22 19]


reshape() → Reshaping

Reshaping permite cambiar la forma de un array sin cambiar sus datos. Es útil cuando necesitas organizar los datos de manera diferente para análisis o visualización.

In [26]:
# Cambiando la forma del array a 2 filas y 3 columnas
edades_reshaped = edades_natacion.reshape(2, 3)
print("Edades reorganizadas:\n", edades_reshaped)

Edades reorganizadas:
 [[20 22 19]
 [21 23 24]]


Funciones Universales (ufuncs)

Suma total de puntos en un torneo
Imaginemos que el Club Deportivo Deporte y Vida organiza un torneo interno de tenis en el que participan varios jugadores, y queremos calcular el total de puntos ganados por todos los jugadores durante el torneo. Cada array representa los puntos ganados por cada jugador en diferentes partidos.

In [27]:
# Puntos ganados por cada jugador en 5 partidos
puntos_jugadores = np.array([
    [15, 12, 9, 7, 6],  # Jugador 1
    [10, 14, 6, 8, 12], # Jugador 2
    [7, 9, 12, 15, 11]  # Jugador 3
])

# Calculamos el total de puntos ganados en el torneo
total_puntos = np.sum(puntos_jugadores)

print(f"El total de puntos ganados por todos los jugadores en el torneo es: {total_puntos}")

El total de puntos ganados por todos los jugadores en el torneo es: 153


Funciones matemáticas
Podríamos calcular el seno de ángulos dados en radianes, lo cual podría ser útil en análisis biomecánicos de movimientos deportivos.

In [28]:
angulos = np.array([0, np.pi/2, np.pi, 3*np.pi/2, 2*np.pi])  # Ángulos en radianes

# Calculando el seno de los ángulos
seno_angulos = np.sin(angulos)

print("Seno de los ángulos:", seno_angulos)

Seno de los ángulos: [ 0.0000000e+00  1.0000000e+00  1.2246468e-16 -1.0000000e+00
 -2.4492936e-16]
